pkg/mcp/source/client_source_test.go

```diff
@@ -134,13 +136,32 @@ func TestClientSource(t *testing.T) {
 		}
 	}
 
-	reconnectChan := make(chan struct{}, 10)
+	waiting := make(chan bool)
```

Comment:
A single buffered channel would work here I think, e.g.

```golang
sync := make(chan struct{}, 1)
reconnectTestProbe = func() { sync <- struct{}{} }

// later 
<-sync
```

Reply:
So, I want the retry loop to stop, then the test code to make some changes, then the retry loop to proceed. I think need two chans for that.